labgrid/resource/udev.py

```diff
@@ -415,3 +415,65 @@ def path(self):
             return self.device.device_path
 
         return None
+
+
+@attr.s(cmp=False)
+class GPIOResource(ManagedResource):
+    manager_cls = UdevManager
+
+    match = attr.ib(default={}, validator=attr.validators.instance_of(dict), hash=False)
+    device = attr.ib(default=None, hash=False)
+    chip = attr.ib(default=None, validator=attr.validators.instance_of(str))
+    offset = attr.ib(default=None, validator=attr.validators.instance_of(int))
+
+    def __attrs_post_init__(self):
+        self.timeout = 5.0
+        self.log = logging.getLogger('GPIOResource')
+        self.match.setdefault('SUBSYSTEM', 'gpio')
+        super().__attrs_post_init__()
+
+    def filter_match(self, device):  # pylint: disable=unused-argument,no-self-use
+        return True
+
+    def try_match(self, device):
+        if self.device is not None:
+            if self.device.sys_path != device.sys_path:
+                return False
+        else: # new device
+            if self.chip != device.sys_name:
+                return False
+            if not self.filter_match(device):
+                return False
+        self.log.debug(" found match: %s", self)
+        if device.action in [None, 'add']:
+            if self.avail:
+                warnings.warn("udev device {} is already available".format(device))
+            self.avail = True
+            self.device = device
+        elif device.action in ['change', 'move']:
+            self.device = device
+        elif device.action in ['unbind', 'remove']:
+            self.avail = False
+            self.device = None
+        self.update()
+        return True
+
+    def update(self):
+        pass
+
+    @property
+    def path(self):
+        if self.device:
+            return str(self.device.sys_name)
+
+        return None
+
+
+@target_factory.reg_resource
+@attr.s(cmp=False)
+class LibGPIOD(GPIOResource):
```

Comment:
As above, I don't see cases where we have different kinds of GPIOResources, so this could be combined.

Reply:
From [my answer above](https://github.com/labgrid-project/labgrid/pull/468#discussion_r301462547):
> I had two scenarios in mind:
> 1. Someone implements the deprecated sysfs interface using SSHDriver
> 2. People running a different OS (bsd/whatever)

Reply:
But I *do* see the obvious point of the unnecessary indirection/abstraction.

Reply:
Before you remove this abstraction, let's first figure out if using libgpiod can reasonably fit the current model (regarding the revert to default on close behaviour).

Reply:
the `udev` variant seized to exist.